pkg/controller/clusterinstallation/controller.go

```diff
@@ -0,0 +1,264 @@
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a ClusterInstallation object and makes changes based on the state read
+// and what is in the ClusterInstallation.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileClusterInstallation) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ClusterInstallation")
+
+	// Fetch the ClusterInstallation instance
+	mattermost := &mattermostv1alpha1.ClusterInstallation{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, mattermost)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	err = mattermost.SetDefaults()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if mattermost.Spec.DatabaseType.Type == "mysql" {
+		reqLogger.Info("Reconciling ClusterInstallation MySQL service account")
+		err = r.checkMySQLServiceAccount(mattermost, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		reqLogger.Info("Reconciling ClusterInstallation MySQL role binding")
+		err = r.checkMySQLRoleBinding(mattermost, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		reqLogger.Info("Reconciling ClusterInstallation MySQL")
+		err = r.checkMySQLDeployment(mattermost, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		reqLogger.Info("Reconciling ClusterInstallation Postgres")
+		err = r.checkDBPostgresDeployment(mattermost, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	// TODO
+	err = r.checkMinioDeployment(mattermost, reqLogger)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = r.checkMattermost(mattermost, reqLogger)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Already exists - don't requeue
+	return reconcile.Result{}, nil
+}
+
+// Object combines the interfaces that all Kubernetes objects must implement.
+type Object interface {
+	runtime.Object
+	v1.Object
+}
+
+// createResource creates the provided resource and sets the owner
+func (r *ReconcileClusterInstallation) createResource(owner v1.Object, resource Object, reqLogger logr.Logger) error {
+	if err := r.client.Create(context.TODO(), resource); err != nil {
```

Comment:
nit: we're deviating from the code above here, and also not being "idiomatic" with respect to error handling here. Perhaps converge on the (slight more verbose):
```
err := r.client.Create(...)
if err != nil {
    // ...
}
```